### PR TITLE
fixed various clippy warnings

### PR DESCRIPTION
**Pull request by llogiq**
Mostly style things, but perhaps it's useful nonetheless. Cheers!

### CODE REVIEW

File: src/ecs/mod.rs [hunk 1]
```diff
@@ -520,7 +520,7 @@ impl ComponentMem {
         let data = self.data[idx].as_ref().unwrap();
         let start = rem * self.component_size;
         unsafe {
-            mem::transmute(data.0.as_ptr().offset(start as isize))
+            &*(data.0.as_ptr().offset(start as isize) as *mut T)
```
Comment (Thinkofname) [inline src/ecs/mod.rs]: Shouldn't this be `*const T`?

Comment (llogiq) [inline src/ecs/mod.rs]: Yeah, sorry that was a copy-paste error.

**Comment by Thinkofname** [comment]
Thanks

**Comment by llogiq** [comment]
My pleasure. I even found a bug in clippy. :smile:

**Pull request merged**